### PR TITLE
Add firefox_kpi_dashboard_v1

### PR DESCRIPTION
**Pull request by jklukas**
This table will be recreated once per day, giving the complete dataset
needed to power the KPI dashboard with CI already populated.

Tableau will be able to hit this table and get cached results,
significantly speeding up the iteration cycle for working with this data
in Tableau.

### CODE REVIEW

**Comment by jklukas** [comment]
This table has proven useful and we're moving forward with implementing a Tableau version of the KPI dashboard on top of it, so this is ready for review and I will be adding it to a DAG.

File: sql/firefox_kpi_dashboard_v1.sql [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+  GROUP BY
+    datasource,
+    type,
+    submission_date ),
+  --
+  with_forecast AS (
+  SELECT
+    datasource,
+    type,
+    submission_date AS date,
```
Comment (relud) [inline sql/firefox_kpi_dashboard_v1.sql]: I think we should quote identifiers that are keywords, even if they aren't [reserved keywords](https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical#reserved-keywords)
```suggestion
    submission_date AS `date`,
```

File: sql/firefox_kpi_dashboard_v1.sql [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+    total AS mau,
+    low AS mau_low,
+    high AS mau_high
+  FROM
+    with_ci
+  UNION ALL
+  SELECT
+    datasource,
+    type,
+    date,
```
Comment (relud) [inline sql/firefox_kpi_dashboard_v1.sql]:
```suggestion
    `date`,
```

File: sql/firefox_kpi_dashboard_v1.sql [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+    type != 'original')
+  --
+SELECT
+  *
+FROM
+  with_forecast
+ORDER BY
+  datasource,
+  type,
+  date
```
Comment (relud) [inline sql/firefox_kpi_dashboard_v1.sql]:
```suggestion
  `date`
```

File: sql/firefox_kpi_dashboard_v1.sql [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+    datasource,
+    type,
+    submission_date,
+    sum_buckets_with_ci(ARRAY_AGG(mau)).*
+  FROM
+    per_bucket
+  GROUP BY
+    datasource,
+    type,
+    submission_date ),
```
Comment (relud) [inline sql/firefox_kpi_dashboard_v1.sql]: this could be an incremental query, change my mind

File: sql/firefox_kpi_dashboard_v1.sql [hunk 5]
```diff
@@ -0,0 +1,231 @@
+CREATE TEMP FUNCTION
+  sum_buckets_with_ci(a ARRAY<INT64>)
```
Comment (relud) [inline sql/firefox_kpi_dashboard_v1.sql]: This seems generic enough to be copied to `udf/udf_sum_buckets_with_ci.js.sql`, and should have a `udf_` prefix for consistency:
```suggestion
  udf_sum_buckets_with_ci(a ARRAY<INT64>)
```

File: sql/firefox_kpi_dashboard_v1.sql [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+  GROUP BY
+    id_bucket,
+    submission_date ),
+  --
+  with_ci AS (
+  SELECT
+    datasource,
+    type,
+    submission_date,
+    sum_buckets_with_ci(ARRAY_AGG(mau)).*
```
Comment (relud) [inline sql/firefox_kpi_dashboard_v1.sql]:
```suggestion
    udf_sum_buckets_with_ci(ARRAY_AGG(mau)).*
```

File: sql/firefox_kpi_dashboard_v1.sql (hunk 4 above)
Comment (jklukas) [reply sql/firefox_kpi_dashboard_v1.sql]: The MAU actuals are all built incrementally, but the forecasts change from day to day. Each day that we recreate this table we will have rows of type "actual" up to yesterday, then we'll have forecast numbers until the end of the year. Tomorrow, a new forecast will run and all those forecast numbers change.

That said, we could add another column to this table that indicates date of running and we could expect that the dashboard always filters for the most recent date, but seems like additional complexity. The point of this table is to make it contain exactly what's needed for presentation on the Tableau dashboard with minimal complexity.

Comment (relud) [reply sql/firefox_kpi_dashboard_v1.sql]: Okay, mind changed, non-incremental seems fine so long as:
1. it runs in a reasonable amount of time
2. it scans a reasonable amount of data

**Pull request merged**